### PR TITLE
Convert productConfigModal to a component

### PR DESCRIPTION
**Pull request by OzzieOrca**
- Move submit button label and translation to template
- Throw error in cortexApiService if path is empty

### CODE REVIEW

File: src/app/productConfig/productConfigModal/productConfig.modal.component.js [hunk 1]
```diff
@@ -183,10 +178,11 @@ class ModalInstanceCtrl {
     savingObservable.subscribe( () => {
       if ( this.isEdit ) {
         this.$scope.$emit( cartUpdatedEvent );
-        this.$uibModalInstance.close( {isUpdated: true} );
+        this.close( {isUpdated: true} );
+        this.close( {isUpdated: false} );
```
Comment (Omicron7) [inline src/app/productConfig/productConfigModal/productConfig.modal.component.js]: Probably don't need 2 this.close()

File: src/app/productConfig/productConfigModal/productConfig.modal.component.js [hunk 2]
```diff
@@ -15,53 +14,49 @@ import desigSrcDirective from 'common/directives/desigSrc.directive';
...
-    } else {
-      this.submitLabel = this.gettext( 'Add to Gift Cart' );
-      this.initializeParams();
-    }
+  $onInit(){
+    this.productData = this.resolve.productData;
+    this.itemConfig = this.resolve.itemConfig;
+    this.isEdit = this.resolve.isEdit;
+    this.uri =  this.resolve.uri;
+    this.suggestedAmounts =  this.resolve.suggestedAmounts;
```
Comment (Omicron7) [inline src/app/productConfig/productConfigModal/productConfig.modal.component.js]: Does this still resolve the dependencies before instantiating the component/controller?

File: src/app/productConfig/productConfigModal/productConfig.modal.component.js (hunk 1 above)
Comment (OzzieOrca) [inline src/app/productConfig/productConfigModal/productConfig.modal.component.js]: That probably came from a merge conflict. I had started this awhile ago. I'll fix it. Good catch.

File: src/app/productConfig/productConfigModal/productConfig.modal.component.js (hunk 2 above)
Comment (OzzieOrca) [inline src/app/productConfig/productConfigModal/productConfig.modal.component.js]: I don't think the component is instantiated until all the resolves are resolved successfully and the modal starts to render.

**Pull request merged**